plugins/org.jkiss.dbeaver.erd.ui/src/org/jkiss/dbeaver/erd/ui/part/DiagramPart.java

```diff
@@ -143,9 +144,15 @@ protected IFigure createFigure()
 
         FanRouter router = new FanRouter();
         router.setSeparation(15);
+        final DBPPreferenceStore store = ERDUIActivator.getDefault().getPreferences();
         //router.setNextRouter(new BendpointConnectionRouter());
-        router.setNextRouter(new ShortestPathConnectionRouter(figure));
-        //router.setNextRouter(new ManhattanConnectionRouter());
+        if (store.getString(ERDUIConstants.PREF_ROUTING_TYPE).equals(ERDUIConstants.ROUTING_MIKAMI)) {
+            router.setNextRouter(new MikamiTabuchiConnectionRouter(figure));
+        } else {
+            router.setNextRouter(new ShortestPathConnectionRouter(figure));
+        }
```

Comment:
I don't like that it's hardcoded just for specific router.

Can we abstract these things into a dedicated "Decorator" class?